### PR TITLE
Add documentation for config.yaml to make the application more user friendly. #97

### LINKED ISSUE #97: Document 'config.yaml' contents
Doesn't do a lot of good to have a `--config CONFIG       Path or URL to the config.yaml file.` call-out in the usage-instructions (and usage document) if there are no indications on what `needs`, should or `can` be in the file.

Lack of this type of documentation makes the software not terribly useful to people that didn't write the software.

### PR DESCRIPTION
**Pull request by MarionTheBull**
Fixes #97

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -1,21 +1,143 @@
...
-point to that file.
+You can create a file using the above format with your own set of standard values and use that file for Watchmaker. Pass the CLI parameter `--config` to point to that file.
 
 [0]: http://www.yaml.org/spec/1.2/spec.html
 [1]: https://github.com/plus3it/watchmaker/blob/develop/src/watchmaker/static/config.yaml
+
+# Config.yaml
+## All
+
+Section for configurations that affect the deployment of the system. All of the configuration parameters can be set in this section. The `all:` section will override parameters that are set in the OS specific sections of the config.yaml, therefor it is suggested practice to break them into the OS blocks of the config. This practice is used in the example config supplied in the next section.
```
Comment (KevinPlus3) [inline docs/configuration.md]: Typo: therefor is missing its ending 'e'.

File: docs/configuration.md [hunk 2]
```diff
@@ -1,21 +1,143 @@
...
+
+
+### Salt
+
+The salt user defined parameters.
```
Comment (KevinPlus3) [inline docs/configuration.md]: Should be a dashL 'user-defined'.

File: docs/configuration.md [hunk 3]
```diff
@@ -1,21 +1,143 @@
...
+
+|Option | Type | Description|
+|-------|:-----:|------------|
+|admin_groups: | (list) | The group(s) that you would like the admin accounts placed.|
```
Comment (KevinPlus3) [inline docs/configuration.md]: Changed to "The group(s) that you would like the admin accounts placed 'in'." (Or 'within' would be acceptable.)

File: docs/configuration.md [hunk 4]
```diff
@@ -1,21 +1,143 @@
...
+| environment: | (boolean) |Set for the environment in which the system is being built. </br> `environment: dev`|
+|formula_termination_strings: | (list) | The defaults are the typical salt formula naming convention. If a user has a salt formula that deviates from the standard naming conventions in Salt, additional strings to match can be defined. </br> Example: Salt default package: `systemprep-formula` </br>User defined package: `systemprep-formula-homemade.zip` </br>The termination string `-homemade` should be added, so that the default formula is overridden properly. |
+| ou_path: | (String) | Specifies the full DN of the OU where the computer account will be created when joining a domain. </br>`ou_path: "OU=Super Cool App, DC=example,DC=com"`|
+| salt_state: | (String) comma seperated | User defined salt states. </br>`salt_state: Highstate, Userstate` |
+|  s3_source: | (string) | Use S3 utilities to retrieve content instead of http(s) utilities. For S3 utilities to work, the system must have boto credentials configured that allow access to the S3 bucket.|
+| user_formulas: | (list) | URL(s) for user defined Salt formulas. |
```
Comment (KevinPlus3) [inline docs/configuration.md]: Again, 'user-defined'.

File: docs/configuration.md [hunk 5]
```diff
@@ -1,21 +1,143 @@
...
+
+|Option | Type | Description|
+|-------|:-----:|------------|
+|admin_groups: | (list) | The group(s) that you would like the admin accounts placed.|
+|admin_users: | (list) |  The user(s) that would be created as admins.  |
+|computer_name: | (string) | The computer or hostname that should be applied. |
+| environment: | (boolean) |Set for the environment in which the system is being built. </br> `environment: dev`|
+|formula_termination_strings: | (list) | The defaults are the typical salt formula naming convention. If a user has a salt formula that deviates from the standard naming conventions in Salt, additional strings to match can be defined. </br> Example: Salt default package: `systemprep-formula` </br>User defined package: `systemprep-formula-homemade.zip` </br>The termination string `-homemade` should be added, so that the default formula is overridden properly. |
+| ou_path: | (String) | Specifies the full DN of the OU where the computer account will be created when joining a domain. </br>`ou_path: "OU=Super Cool App, DC=example,DC=com"`|
+| salt_state: | (String) comma seperated | User defined salt states. </br>`salt_state: Highstate, Userstate` |
```
Comment (KevinPlus3) [inline docs/configuration.md]: 'User-defined'

File: docs/configuration.md [hunk 6]
```diff
@@ -1,21 +1,143 @@
...
+|  s3_source: | (string) | Use S3 utilities to retrieve content instead of http(s) utilities. For S3 utilities to work, the system must have boto credentials configured that allow access to the S3 bucket.|
+| user_formulas: | (list) | URL(s) for user defined Salt formulas. |
+
+## Linux
+Section for configurations that affect the deployment of the system that are only applicable to Linux based systems.
```
Comment (KevinPlus3) [inline docs/configuration.md]: 'Linux-based'

File: docs/configuration.md [hunk 7]
```diff
@@ -1,21 +1,143 @@
...
+
+### Yum
+
+|Option | Type | Description|
+|------|:----:|-----|
+|repo_map: | Add Section |There be dragons here!</br>Please becareful making changes to the existing repos unless you are sure of what you are doing. You can add repos here that you would like enabled but be weary of changing the existing. Select the EL version that an additional repo should be added to. Then add a full block, redhat and centos are the only supported 'enterprise linux' distros currently supported. Must include dist, el_version, and url|
```
Comment (KevinPlus3) [inline docs/configuration.md]: Not sure what "Add Section" means especially in context of rest of configurations. I assume "There be dragons here!" should be removed unless left for humorous purposes? Need a space in 'becareful'. Another change: add 'repos' to end of sentence that starts with 'You can add...'

File: docs/configuration.md [hunk 8]
```diff
@@ -1,21 +1,143 @@
...
+  url: http://someplace.com/my.repo
+```
+
+### Salt
+
+Salt configurations that pertain to linux operating system configurations.
+
+|Option | Type | Description|
+|-------|:-----:|------------|
+|salt_debug_log:| (String) | Path to the debug logfile that should be used for Salt.|
```
Comment (KevinPlus3) [inline docs/configuration.md]: Not sure about this: should there be a space in 'logfile'?

Comment (KevinPlus3) [inline docs/configuration.md]: Never mind. Looked it up and single word is fine.

File: docs/configuration.md (hunk 7 above)
Comment (MarionTheBull) [inline docs/configuration.md]: Add Section should be 'Add Entire Block'

Humorous purposes to mean be careful mucking with the portions that are already defined.

other suggestions accepted.

File: docs/configuration.md (hunk 6 above)
Comment (MarionTheBull) [inline docs/configuration.md]: done.

File: docs/configuration.md (hunk 5 above)
Comment (MarionTheBull) [inline docs/configuration.md]: done.

File: docs/configuration.md (hunk 4 above)
Comment (MarionTheBull) [inline docs/configuration.md]: done.

File: docs/configuration.md (hunk 3 above)
Comment (MarionTheBull) [inline docs/configuration.md]: done.

File: docs/configuration.md (hunk 2 above)
Comment (MarionTheBull) [inline docs/configuration.md]: done.

**Pull request merged**